script.js

```diff
@@ -1,4 +1,190 @@
+// in order to simplify our code, we assume that once the game mode has been selected, it cannot be switched.
+// also assume that the user knows the rules of each mode so we don't have to add explanation text.
+
+// constants
+var NUMBER_OF_CHOICES = 3;
+var SCISSORS = "scissors";
+var PAPER = "paper";
+var STONE = "stone";
+var PENDING_USER_NAME = "waiting for user name";
+var PENDING_GAME_MODE = "pending game mode";
+var NORMAL = "normal";
+var REVERSE = "reverse";
+var COM_VS_COM = "computer vs computer";
+var KOREAN = "korean";
+var WINNING_PCT_THRESHOLD = 33;
+var USER_WIN = "you";
+var DRAW = "draw";
+var COM_WIN = "the computer";
+var NO_WINNER_YET = "no winner yet";
+var PENDING_DRAW = "pending draw";
+
+// global variables
+var numUserWins = 0;
+var numComWins = 0;
+var numDraws = 0;
+var numGames = 0;
+var gameMode = PENDING_USER_NAME; // initial game mode
+var userName;
+var userChoice;
+var progChoice;
+var gameResult;
+var latestWinner = NO_WINNER_YET;
+
+var generateRandomChoice = function () {
+  var randomDecimal = Math.random() * NUMBER_OF_CHOICES;
+  var randomInteger = Math.floor(randomDecimal) + 1; // possible values = 1,2,3
+  var randomChoice = SCISSORS;
+  if (randomInteger == 1) {
+    randomChoice = PAPER;
+  }
+  if (randomInteger == 2) {
+    randomChoice = STONE;
+  }
+  return randomChoice;
+};
+
+var createEmoji = function (choice) {
+  var emoji;
+  if (choice == SCISSORS) {
+    emoji = "✂️";
+  } else if (choice == PAPER) {
+    emoji = "📄";
+  } else if (choice == STONE) {
+    emoji = "🗿";
+  }
+  return emoji;
+};
+
+var basicSPS = function (userMove) {
+  progChoice = generateRandomChoice();
+  if (gameMode == COM_VS_COM) {
+    userChoice = generateRandomChoice();
+  } else {
+    userChoice = userMove;
+  }
+  if (userChoice == progChoice) {
+    gameResult = DRAW;
+  } else if (
+    (userChoice == SCISSORS && progChoice == STONE) ||
+    (userChoice == PAPER && progChoice == SCISSORS) ||
+    (userChoice == STONE && progChoice == PAPER)
```

Comment:
This logic seems to be repeated several times in your code. This is a tell tale sign that you could consider abstracting it into a separate function.